test/dreyfus_purge_test.erl

```diff
@@ -0,0 +1,543 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(dreyfus_purge_test).
+
+-include_lib("couch/include/couch_db.hrl").
+-include_lib("dreyfus/include/dreyfus.hrl").
+-include_lib("couch/include/couch_eunit.hrl").
+
+-export([test_purge_single/0, test_purge_multiple/0, test_purge_multiple2/0,
+    test_purge_conflict/0, test_purge_conflict2/0, test_purge_update/0, test_purge_update2/0,
+    test_delete/0, test_delete_purge_conflict/0, test_delete_conflict/0,
+    test_all/0]).
+%-export([create_db_docs/1, create_docs/2, delete_db/1, purge_one_doc/2, dreyfus_search/2]).
+
+test_all() ->
+    test_purge_single(),
+    test_purge_multiple(),
+    test_purge_multiple2(),
+    test_purge_conflict(),
+    test_purge_update(),
+    test_purge_update2(),
+    test_purge_conflict2(),
+    test_delete(),
+    test_delete_purge_conflict(),
+    test_delete_conflict(),
+    ok.
+
+test_purge_single() ->
+    DbName = db_name(),
+    create_db_docs(DbName),
+    {ok, _, HitCount1, _, _, _} = dreyfus_search(DbName, <<"apple">>),
+    ?assertEqual(HitCount1, 1),
+    DBFullName = get_full_dbname(DbName),
+    ok = purge_one_doc(DBFullName, <<"apple">>),
+    {ok, _, HitCount2, _, _, _} = dreyfus_search(DbName, <<"apple">>),
+    ?assertEqual(HitCount2, 0),
+    delete_db(DbName),
+    ok.
+
+test_purge_multiple() ->
+    Query = <<"color:red">>,
+
+    %create the db and docs
+    DbName = db_name(),
+    create_db_docs(DbName),
+
+    %first search request
+    {ok, _, HitCount1, _, _, _} = dreyfus_search(DbName, Query),
+
+    ?assertEqual(HitCount1, 5),
```

Comment:
this could be improved with `?assertMatch`, i.e;

`?assertMatch({ok, _, 5, _, _, _}, dreyfus_search(DbName, Query)),`